File: solutions/4129-number-of-centered-subarrays/number-of-centered-subarrays.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int centeredSubarrays(int[] nums) {
        int a = 0, b = nums.length;
        int[] c = nums;
        for (int d = 0; d < b; ++d) {
            int e = 0;
            Set<Integer> f = new HashSet<>();
            for (int g = d; g < b; ++g) {
                e += c[g];
                f.add(c[g]);
                if (f.contains(e)) a++;
            }
        }
        return a;
    }
}

File: solutions/4129-number-of-centered-subarrays/number-of-centered-subarrays.py
class Solution:
    def centeredSubarrays(self, nums: List[int]) -> int:
        result = 0
        n = len(nums)
        for start in range(n):
            running_sum = 0
            elements_set = set()
            for end in range(start, n):
                running_sum += nums[end]
                elements_set.add(nums[end])
                if running_sum in elements_set:
                    result += 1
        return result